File: core/src/galenscovell/oregontrail/things/inanimate/Location.java
package galenscovell.oregontrail.things.inanimate;

import galenscovell.oregontrail.map.Sector;
import galenscovell.oregontrail.things.inanimate.Events.Event;
import galenscovell.oregontrail.util.Repository;

import java.util.*;

public class Location {
    public final int x, y, size;
    private Sector sector;
    private ArrayList<Event> events;
    private String[] details;

    public Location(int x, int y, int size) {
        this.x = x;
        this.y = y;
        this.size = size;
        this.details = new String[]{"Location Title", "Location Detail"};
    }

    public void setSector(Sector sector) {
        this.sector = sector;
        sector.becomeUnexplored();
    }

    public Sector getSector() {
        return sector;
    }

    public void enter() {
        generateEvents(new Random());
    }

    public String[] getDetails() {
        return details;
    }

    private void generateEvents(Random random) {
        // Each Location has between 1 and 4 events
        this.events = new ArrayList<Event>();
        int numberOfEvents = random.nextInt(4) + 1;
        for (int event = 0; event < numberOfEvents; event++) {

        }
        // Eventually background should be consistent and not created here
        createBackground(random);
    }

    private void createBackground(Random random) {
        // Background depends on number and type of events generated
        // eg many planet events = background has planets
        // eg no planet events = background has no planets
        int num = random.nextInt(4);
        String layerName;
        if (num == 0) {
            layerName = "blue_bg";
        } else if (num == 1) {
            layerName = "purple_bg";
        } else if (num == 2) {
            layerName = "green_bg";;
        } else {
            layerName = "";
        }
        Repository.gameScreen.setBackground(layerName, "bg1", "bg2", layerName, "bg1_blur", "bg2_blur");
    }
}

File: core/src/galenscovell/hinterstar/things/inanimate/Location.scala
package galenscovell.hinterstar.things.inanimate

import java.util._

import galenscovell.hinterstar.map.Sector
import galenscovell.hinterstar.things.inanimate.Events.{Event, Planet}
import galenscovell.hinterstar.util.{Constants, Repository}

import scala.collection.mutable.ArrayBuffer


class Location(xIn: Int, yIn: Int, sizeIn: Int) {
  final val x: Int = xIn
  final val y: Int = yIn
  final val size: Int = sizeIn
  private val events: ArrayBuffer[Event] = ArrayBuffer[Event]()
  private val details: Array[String] = Array[String]("Location Title", "Location Detail")
  private var sector: Sector = null


  def setSector(sectorIn: Sector): Unit = {
    this.sector = sectorIn
    sector.becomeUnexplored()
  }

  def getSector: Sector = {
    sector
  }

  def enter(): Unit = {
    generateEvents(new Random)
  }

  def getDetails: Array[String] = {
    details
  }

  def getEvents: ArrayBuffer[Event] = {
    events
  }

  private def generateEvents(random: Random): Unit =  {
    // Each Location has between 3 and 6 events
    this.events.clear()
    val numberOfEvents: Int = random.nextInt(4) + 3

    // Randomly set distances between events
    val distances: ArrayBuffer[Int] = new ArrayBuffer[Int]()
    val baseDistance: Float = (Constants.PROGRESS_PANEL_WIDTH - (numberOfEvents * 3f)) / numberOfEvents
//    while (distances.size < numberOfEvents) {
//      val randomDifference: Int = random.nextInt(30) - 15
//      distances += (baseDistance + randomDifference)
//    }

    for (x <- 0 until numberOfEvents) {
//      events += new Planet(distances(x))
      events += new Planet(baseDistance)
    }
    // Eventually background should be consistent and not created here
    createBackground(random)
  }

  private def createBackground(random: Random): Unit = {
    // Background depends on number and type of events generated
    // eg many planet events = background has planets
    // eg no planet events = background has no planets
    val num: Int = random.nextInt(4)
    var layerName: String = null

    if (num == 0) {
      layerName = "blue_bg"
    }
    else if (num == 1) {
      layerName = "purple_bg"
    }
    else if (num == 2) {
      layerName = "green_bg"
    }
    else {
      layerName = ""
    }
    Repository.gameScreen.setBackground(layerName, "bg1", "bg2", layerName, "bg1_blur", "bg2_blur")
  }
}
